daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala

```diff
@@ -107,6 +107,47 @@ object ParticipantsJsonProtocol extends DefaultJsonProtocol {
   implicit val participantsFormat = jsonFormat3(Participants[ApiParameters])
 }
 
+final case class Script(id: Identifier, expr: SExpr, param: Option[Type], scriptIds: ScriptIds)
+
+object Script {
+  def fromDar(dar: Dar[(PackageId, Package)], scriptId: Identifier): Script = {
+    val darMap = dar.all.toMap
+    val compiler = Compiler(darMap)
+    val compiledPackages =
+      PureCompiledPackages(darMap, compiler.compilePackages(darMap.keys)).right.get
+    fromIdentifier(compiledPackages, scriptId)
+  }
+  def fromIdentifier(compiledPackages: CompiledPackages, scriptId: Identifier): Script = {
+    val scriptExpr = SEVal(LfDefRef(scriptId), None)
+    val scriptTy = compiledPackages
+      .getPackage(scriptId.packageId)
+      .flatMap(_.lookupIdentifier(scriptId.qualifiedName).toOption) match {
+      case Some(DValue(ty, _, _, _)) => ty
+      case Some(d @ DTypeSyn(_, _)) =>
+        throw new RuntimeException(s"Expected DAML script but got synonym $d")
```

Comment:
It probably makes sense to also change this to return `Either[String, Script]` similar to the trigger runner.

Reply:
Yes, it turns out this allows to simplify the `unapply` bit in `TestMain`. I've updated the PR.